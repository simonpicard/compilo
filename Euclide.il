a::Integer
b::Integer
c::Integer

# Euclidean algorithm

function main()
	a = readint()
	b = readint()
	while(b)
		c=b;
		while(a >= b) # a mod b
			a=a-b
		end
		b=a;a=c
	end
	println(a) 
end

main()
return
