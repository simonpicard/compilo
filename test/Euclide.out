token: a	lexical unit: IDENTIFIER
token: ::	lexical unit: TYPE_DEFINITION
token: Integer	lexical unit: INTEGER_TYPE
token: b	lexical unit: IDENTIFIER
token: ::	lexical unit: TYPE_DEFINITION
token: Integer	lexical unit: INTEGER_TYPE
token: c	lexical unit: IDENTIFIER
token: ::	lexical unit: TYPE_DEFINITION
token: Integer	lexical unit: INTEGER_TYPE
token: function	lexical unit: FUNCTION
token: main	lexical unit: IDENTIFIER
token: (	lexical unit: LEFT_PARENTHESIS
token: )	lexical unit: RIGHT_PARENTHESIS
token: a	lexical unit: IDENTIFIER
token: =	lexical unit: ASSIGNATION
token: readint	lexical unit: READ_INTEGER
token: (	lexical unit: LEFT_PARENTHESIS
token: )	lexical unit: RIGHT_PARENTHESIS
token: b	lexical unit: IDENTIFIER
token: =	lexical unit: ASSIGNATION
token: readint	lexical unit: READ_INTEGER
token: (	lexical unit: LEFT_PARENTHESIS
token: )	lexical unit: RIGHT_PARENTHESIS
token: while	lexical unit: WHILE
token: (	lexical unit: LEFT_PARENTHESIS
token: b	lexical unit: IDENTIFIER
token: )	lexical unit: RIGHT_PARENTHESIS
token: c	lexical unit: IDENTIFIER
token: =	lexical unit: ASSIGNATION
token: b	lexical unit: IDENTIFIER
token: ;	lexical unit: END_OF_INSTRUCTION
token: while	lexical unit: WHILE
token: (	lexical unit: LEFT_PARENTHESIS
token: a	lexical unit: IDENTIFIER
token: >=	lexical unit: GREATER_OR_EQUALS_THAN
token: b	lexical unit: IDENTIFIER
token: )	lexical unit: RIGHT_PARENTHESIS
token: a	lexical unit: IDENTIFIER
token: =	lexical unit: ASSIGNATION
token: a	lexical unit: IDENTIFIER
token: -	lexical unit: MINUS
token: b	lexical unit: IDENTIFIER
token: end	lexical unit: END
token: b	lexical unit: IDENTIFIER
token: =	lexical unit: ASSIGNATION
token: a	lexical unit: IDENTIFIER
token: ;	lexical unit: END_OF_INSTRUCTION
token: a	lexical unit: IDENTIFIER
token: =	lexical unit: ASSIGNATION
token: c	lexical unit: IDENTIFIER
token: end	lexical unit: END
token: println	lexical unit: PRINTLN
token: (	lexical unit: LEFT_PARENTHESIS
token: a	lexical unit: IDENTIFIER
token: )	lexical unit: RIGHT_PARENTHESIS
token: end	lexical unit: END
token: main	lexical unit: IDENTIFIER
token: (	lexical unit: LEFT_PARENTHESIS
token: )	lexical unit: RIGHT_PARENTHESIS
Identifiers
a	1
b	2
c	3
main	7
